src/export_from_inboveg/HT31xx_data_preparation.Rmd

```diff
@@ -0,0 +1,400 @@
+
+```{r}
+site_characteristics <- bind_rows(structure, layerqual) %>%
+    mutate(var = str_to_lower(var),
+           var = str_remove(var, 
+                             paste(c(" \\(\\%\\)", " in meter", " in m"), 
+                                   collapse = "|")),
+           value = str_to_lower(value)) 
+# to do (later): toevoegen van beschrijving van klassen?
+
+```
+
+```{r}
+site_characteristics %>% head() %>% kable() %>% kable_styling() %>% scroll_box(width = "800px")
+```
+
+
+## vegetation data
+
+lijst van 'my taxa' (niet in inboveg-lijsten)
+```{r}
+vegetation_31xx %>% 
+  filter(is.na(name_scientific)) %>% 
+  distinct(name_original) %>% 
+  arrange(name_original)
+```
+
+rearrangement of vegetation
+```{r}
+vegetation <- vegetation_31xx %>% 
+  mutate(name = ifelse(is.na(name_scientific), name_original, name_scientific),
+         name = case_when(
+           name_original %in% c("draadwier", "Draadwier spec.") ~ "Draadwier",
+           name_original %in% 
+             c("Hydrodictyon reticulatum (Linnaeus) Bory", 
+               "Hydrodictyon reticulatum Lagerh.") ~ "Hydrodictyon reticulatum",
+           name_original %in% 
+             c("kroossoorten", 
+               "kroossoorten (= Lemna sp., Spirodela polyrhiza, Wolffia arrhiza Uitz. Lemna trisulca)") ~ "Lemna",
+            TRUE ~ name_original)) %>% 
+  # synoniemen van my taxa oplossen & deze veranderen zodat deze leesbaar zijn door LSVI-rekenmodule 
+  select(survey, recording_givid, name, phenology_code, species_cover_code, 
+         species_cover, scale) %>% 
+  distinct()  # sommige soorten staan er anders dubbel in omdat ze aan verschillende verstoringslagen zijn gekoppeld: Agrostis canina, ...
+
+```
+
+Sommige taxa zijn een groepering van soorten en hebben geen wetenschappelijke naam in INBOVEG. Voor de ontbrekende namen bij name_scientific, wordt de name_original overgenomen in de 'name'.
+
+```{r}
+vegetation %>% head() %>% kable() %>% kable_styling() %>% scroll_box(width = "800px")
+```
+
+# data controle 
+
+##  input
+
+```{r, eval = FALSE}
+this_sample <- "LI_HEU_004_3160_20"
+
+header_sel <- header_31xx %>% filter(str_detect(user_reference, this_sample))
+
+classif_sel <- classif_31xx  %>% semi_join(header_sel, by = "recording_givid")
+structure_sel <- structure_31xx  %>% semi_join(header_sel, by = "recording_givid")
+sitequal_sel <- sitequal_31xx  %>% semi_join(header_sel, by = "recording_givid")
+layerqual_sel <- layerqual_31xx  %>% semi_join(header_sel, by = "recording_givid")
+vegetation_sel <- vegetation_31xx  %>% semi_join(header_sel, by = "recording_givid")
+```
+
+ziet er ok uit  
+
+##  output
+
+```{r, eval = FALSE}
+header_sel <- header %>%  
+  filter(str_detect(user_reference, this_sample))
+site_characteristics_sel <- site_characteristics %>% 
+  semi_join(header_sel, by = "recording_givid")
+vegetation_sel <- vegetation %>% 
+  semi_join(header_sel, by = "recording_givid")
+```
+
+ziet er ok uit
+
```

Comment:
```suggestion
doublerecords_site_characteristics <- site_characteristics %>%
  group_by(recording_givid, var, value) %>%
  mutate(n_records = n()) %>%
  ungroup() %>%
  filter(n_records > 1)
  
  doublerecords_vegetation <- vegetation %>%
  group_by(recording_givid, name, phenology_code) %>%
  mutate(n_records = n()) %>%
  ungroup() %>%
  filter(n_records > 1)
```

Reply:
There are several double records for site_characterisics. You van check them with the code above. It is also the reason why you get a warning when using write_vc.

There also som some combinations of recording_givid, name, phenology_code with two cover codes. You noticed it in the data exploration, but you could also check it here.

Reply:
If you press commit suggestion the suggested code is added in this brach, and you can run it in RStudio after you apply do a pull.